scripts/tools/FileSerializerMigration.php

```diff
@@ -0,0 +1,219 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017  (original work) Open Assessment Technologies SA;
+ *
+ * @author Martijn Swinkels <m.swinkels@taotesting.com>
+ */
+
+namespace oat\generis\scripts\tools;
+
+use common_Exception;
+use common_exception_Error;
+use common_report_Report as Report;
+use common_report_Report;
+use oat\generis\Helper\FileSerializerMigrationHelper;
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+
+/**
+ * Migrate ResourceFileSerializer references to the new UrlFileSerializer system
+ *   [properties]
+ *      --wet-run (-w) - Execute the migration
+ *      --limit (-l) - The amount of resources that are processed in one batch
+ */
+class FileSerializerMigration extends ScriptAction
+{
+    const RESOURCE_LIMIT = 200;
+    const MIGRATION_REPORT_LINES = [
+        'migration_success' => [
+            'dry' => '%s old references will be migrated into %s new file serializer references.',
+            'wet' => 'Successfully migrated %s old references to %s new file serializer references.'
+        ],
+        'migration_errors' => [
+            'dry' => '%s resources will fail to migrate. Please verify the following resources:',
+            'wet' => 'Unable to migrate %s resources. Please verify the following resources:'
+        ],
+        'serializer_update_success' => [
+            'dry' => 'FileReferenceSerializer service will be updated to use UrlFileSerializer service.',
+            'wet' => 'Successfully updated FileReferenceSerializer service to use UrlFileSerializer service.'
+        ],
+    ];
+
+    /**
+     * @var FileSerializerMigrationHelper
+     */
+    private $migrationHelper;
+
+    /**
+     * @var Report
+     */
+    private $report;
+
+    /**
+     * Run the script.
+     *
+     * @return common_report_Report
+     * @throws InvalidServiceManagerException
+     * @throws common_Exception
+     * @throws common_exception_Error
+     */
+    protected function run()
+    {
+        $this->report = Report::createInfo('Starting file serializer migration.');
+
+        $migrationHelper = $this->getMigrationHelper();
+
+        $migrationHelper->migrateFiles(
+            $this->hasOption('resourceLimit') ? $this->getOption('resourceLimit') : self::RESOURCE_LIMIT
```

Comment:
this is called resourceLimit here, but gets ibnterpreted as cache size in the Helper, could be confussing